gato/tasks/text_task.py

```diff
@@ -0,0 +1,78 @@
+# supports dataset in huggingface datasets library for now
+from datasets import load_dataset, concatenate_datasets
+from gato.tasks.task import Task, TaskTypeEnum
+import numpy as np
+import math
+from torch.nn import functional as F
+from typing import List
+
+class TextTask(Task): 
+       
+    def __init__(self, task_type, dataset_names:List[str]):
+        super().__init__(task_type)
+        text_datasets_list = []
+        for text_dataset in dataset_names:
+            text_datasets_list.append(load_dataset(path='wikitext', name=text_dataset))
+        if len(text_datasets_list) == 1:
```

Comment:
The text task looks good so far, I am curious about the choice / what you think is the best way to handle having multiple  datasets. Are there speed benefits following the process here, of concatenating the datasets? If we had separate tasks, then we would also want to calculate the total tokens for each task / proportionally calculate how much of each batch comes from each task depending on # of tokens, but we don't have to worry about this if following your procedure. It seems like there is an edge case where the concatenation will not work if the columns are not named the same: https://huggingface.co/docs/datasets/process#concatenate

One thing that may be useful, is that if we have multiple datasets which are concatenated, is during evaluation, is to determine specific metrics associated w/ each separate dataset. E.g., want a separate perplexity score for wikitext vs the pile, not just the average between both. Potentially, after contenating, can maybe maintain start and end indices for each dataset, e.g. pile is 0 to 200mil, other dataset is (200mil + 1) to 400mil, so we can attribute which samples correspond to each task, separately aggregate their metrics.

Reply:
All valid points. Current strategy is simple.

- "there is an edge case where the concatenation will not work if the columns are not named the same" Yes but most good/large datasets do have them following the names. 
- "E.g., want a separate perplexity score for wikitext vs the pile, not just the average between both" - I'll raise a separate issue for this. I don't think this would be required anytime soon. We will switch to Pile once we are able to run on large scale data, and stick to only wikitext until then. For now, I imagine only one text dataset to be there. The first approach you mentioned here is better.

Reply:
Added #22.